source/virtualBuffers/gecko_ia2.py

```diff
@@ -56,16 +56,13 @@ def _getOffsetFromPoint(self,x,y):
 		if not obj or not hasattr(obj, "IAccessibleTextObject"):
 			raise LookupError
 		# Find out whether this object contains text that has been stripped in the virtual buffer.
-		for field in reversed(self._getFieldsInRange(start, start+1)):
+		relOffset = IA2TextTextInfo._getOffsetFromPointInObject(obj, x, y)
+		strippedCharsFromStart = 0
+		for field in reversed(self._getFieldsInRange(start, min(end-1, start+relOffset))):

```

Comment:
This method could still break... and it's actually worse than I originally thought; it affects more than just stripped char searching. You're assuming the text offsets in the buffer match the IA2 text offsets. This won't be true in several cases:

1. The target text has an embedded object before it. In that case, the embedded object will consume one IA2 offset, but in the buffer, the content of the embedded object will be rendered. For example:

    `data:text/html,<p><a href="https://mozilla.org">link</a>text</p>`

    Imagine your mouse is over the first letter "t" in text. That's IA2 offset 1 in the paragraph, but in the buffer, that's actually offset 4.

2. The target object isn't in the buffer; e.g. listbox items or subtree overridden by aria-label. You get around this by walking the object's parents... but now you might be dealing with a completely different object, and thus the offsets are no longer referring to the same thing. Example:

    `data:text/html,<h1 aria-label="h1"><a href="https://mozilla.org">link</a></h1>`

    If your mouse is on the "n" of link, that's IA2 offset 2, but that character doesn't even exist in the buffer, nor does the offset.

So:

1. You need to be looking at the IA2 offsets in the buffer (and the stripped chars), as well as restricting the search to the same object.
2. If the object isn't in the buffer (aside from text leaf nodes where going parent is safe), you should probably just return the first offset for the nearest ancestor.

Reply:
Hmm, it turns out that this code is quite hard to test anyway. I'm leaning towards removing it for now, it is not strictly necessary for virtual buffers to get offsets from point, as mouse echo functions at the object level and doesn't deal with tree interceptors.